### PR TITLE
Added on_behalf_of header.

### PR DESCRIPTION
**Pull request by mattronix**
This header allows for API requests on behalf of a user when using token auth.

### CODE REVIEW

**Comment by joeirimpan** [comment]
@mattronix Can you fix the flake8 issue? Also it would be nice to have a property `on_behalf_of` getter, setter for `_on_behalf_of`.

**Comment by mattronix** [comment]
Will try :)

**Comment by mattronix** [comment]
@joeirimpan silly question how do i fix the test:

./build/lib/zammad_py/api.py:18:80: E501 line too long (98 > 79 characters)

im a bit new so if you can tell me where its going wrong il fix :)

File: zammad_py/api.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 class ZammadAPI(object):
 
     def __init__(
-            self, username, password, host, http_token=None, oauth2_token=None,
+            self, username, password, host, http_token=None, oauth2_token=None, on_behalf_of=None,
```
Comment (joeirimpan) [inline zammad_py/api.py]: newline after oauth2_token=None,

Comment (joeirimpan) [reply zammad_py/api.py]: on_behalf_of=None, is_secure=True

File: zammad_py/api.py [hunk 2]
```diff
@@ -33,6 +34,9 @@ def __init__(
         if self._http_token:
             self.session.headers['Authorization'] = \
                 'Token token=%s' % self._http_token
+        elif self._on_behalf_of:
```
Comment (joeirimpan) [inline zammad_py/api.py]:
```python
if self._http_token:
    self.session.headers['Authorization'] = \
        'Token token=%s' % self._http_token
elif oauth2_token:
    self.session.headers['Authorization'] = \
        'Bearer %s' % self._oauth2_token
else:
    self.session.auth = (self._username, self._password)

if self._on_behalf_of:
    self.session.headers['X-On-Behalf-Of'] = \
        self._on_behalf_of
```

**Comment by mattronix** [comment]
> @joeirimpan silly question how do i fix the test:
> 
> ./build/lib/zammad_py/api.py:18:80: E501 line too long (98 > 79 characters)
> 
> im a bit new so if you can tell me where its going wrong il fix :)

Fixed it i was being silly and did not think to just move it a little.

File: zammad_py/api.py (hunk 2 above)
Comment (joeirimpan) [reply zammad_py/api.py]: Add two more methods to the class
```python
    @property
    def on_behalf_of(self):
        return self._on_behalf_of
    
    @on_behalf_of.setter
    def on_behalf_of(self, value):
        self._on_behalf_of = value
```

Comment (mattronix) [reply zammad_py/api.py]: Added, looks correct to you?

Comment (joeirimpan) [reply zammad_py/api.py]: thanks

**Pull request merged**